### PR TITLE
CPDP-1900: added period of admin option

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

**Comment by AndyHWChung** [comment]
one of the main things with the FE PR one is the introduction of the RepresentativeType type trait which extends IndividualType - I've replaced all the Either[PersonalRep,Capacitor]  with RepresentativeType and made period of admin behave exactly like the existing personal rep stuff for now. All the machinery should be there now to update the content for period of admin throughout the journey.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html [hunk 1]
```diff
@@ -42,16 +43,20 @@
...
 @isAgent = @{ request.userType.contains(UserType.Agent) }
 @userKey = @{
-  if (representativeType.exists(_.isLeft)) ".personalRep"
-  else if (representativeType.exists(_.isRight)) ".capacitor"
-  else if (isAgent) ".agent"
-  else if (isATrust) ".trust"
-  else ""
+  representativeType match {
+      case Some(PersonalRepresentative) => ".personalRep"
+      case Some(PersonalRepresentativeInPeriodOfAdmin) => ".personalRep"
```
Comment (rambabu-posa) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html:53]: Are we going to use same "PersonalRepresentative" messages for "PersonalRepresentativeInPeriodOfAdmin"?

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html [hunk 2]
```diff
@@ -74,11 +79,17 @@
...
-    else if(request.userType.exists(_ === Agent))   messages("address.uk.companyDetails.agent.title")
-    else                                            messages("address.uk.companyDetails.title")
+    else
+        c.representativeType match {
+            case Some(PersonalRepresentative) => messages("address.uk.companyDetails.personalRep.title")
+            case Some(PersonalRepresentativeInPeriodOfAdmin) => messages("address.uk.companyDetails.personalRep.title")
+            case Some(Capacitor) => messages("address.uk.companyDetails.capacitor.title")
+            case None =>
+                if(c.isATrust)                                  messages("address.uk.companyDetails.trust.title")
+                else if(request.userType.exists(_ === Agent))   messages("address.uk.companyDetails.agent.title")
```
Comment (rambabu-posa) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html:89]: Is it possible to use existing "@isAgent" reference here?

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/isUk.scala.html [hunk 3]
```diff
@@ -43,12 +44,17 @@
...
-      else                                            ".companyDetails."
+        if(c.draftReturn.isLeft)                       ".multipleIndirect."
+        else
+            c.representativeType match {
+                case Some(PersonalRepresentative) => ".companyDetails.personalRep."
+                case Some(PersonalRepresentativeInPeriodOfAdmin) => ".companyDetails.personalRep."
+                case Some(Capacitor) => ".companyDetails.capacitor."
+                case None =>
+                    if (c.isATrust) ".companyDetails.trust."
+                    else if (request.userType.exists(_ === Agent)) ".companyDetails.agent."
```
Comment (rambabu-posa) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/isUk.scala.html:55]: As we are doing this logic twice, is it good to extract this into a variable
@isAgent = @{ request.userType.contains(UserType.Agent) }

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html (hunk 1 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html:53]: no, but for now that's what it's going to be until we do the tickets to implement it properly for period of admin.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/isUk.scala.html (hunk 3 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/isUk.scala.html]: yeah, that's a good idea. I've factored it out now

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html (hunk 2 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/enter_uk_address.scala.html]: done

**Comment by rambabu-posa** [comment]
LGTM

**Pull request merged**